http/http.go

```diff
@@ -75,16 +139,14 @@ func PostHelper(ctx context.Context, httpClient *http.Client, stats *statsd.Clie
 	if compress {
 		req.Header.Set("Content-Encoding", "deflate")
 	}
-	// we only make http requests at flush time, so keepalive is not a big win
-	req.Close = true
 
```

Comment:
If we turn off connection closing, we should specify a max keepalive connection lifetime; see https://github.com/stripe/veneur/pull/326/files#diff-34c6b408d72845d076d47126c29948d1R155 for how I did it (I think that one is ~about right)

Reply:
Aaaalso, I'm not sure if we ought to unconditionally not-close connections? There were some that might benefit from being re-opened all the time; not sure that's really true though.

Reply:
Yes! One change though, the `DefaultMaxIdleConnsPerHost` for Go is currently `2` and because we max multiple concurrent connections when we flush to Datadog, that's probably better than `1`. So unless you disagree I think I'll leave that?

Reply:
I can't think of any of our connections (forward, datadog, etc) that would benefit from closing. The one (mild) concern I do have is that this will create _affinity_ from individual instances to a specific outbound proxy or global instance. We miiiight want to periodically close — I wish Go had a "maximum connection lifetime" parameter — to force instances to find a new target. But I'm not worried about hosts going down since that will force a reconnect.

Reply:
Yep, we should definitely experiment with the number of idle conns! I think any number >=1 and <= 2* our concurrency for submission should be fine.

Reply:
Wait, go totally has a max idle connection lifetime parameter on Transport!! `IdleConnTimeout` (:

Reply:
Yeah, but that's how long it's _idle_ not how long it's alive in total, right?

Reply:
That's true. So, one thing that I'm wondering about (and which is why I set it to 1 initially) is if `MaxIdleConnsPerHost` counts per-IP, or per-name. If it's per-IP, we probably want to set that to 1, so that we tend to mint new connections rather than re-using possibly stale / host-affine ones. Not sure if that's exactly the effect we'll get, though.

Reply:
It's per hostname, not per IP. I had to trace through the code to discern that and it's possible I'm wrong.

We only "break up" Datadog POSTs to multiple requests, so we shouldn't really end up with more than one for other things.

We get one new metric from this — ignoring the spans — which is the number of connections we open tagged by state of `new` or `reused`. This will let us learn in practice what the ratio is. We can run this in QA for a while and adjust?